src/options.cpp

```diff
@@ -0,0 +1,408 @@
+    sock_str << ":" << options["port"].as<int>();
+    socket_path = sock_str.str();
+  }
+  if (options.count("memcache"))
+    memcache_spec = options["memcache"].as<std::string>();
+  if (options.count("ratelimit"))
+    rate_limit = options["ratelimit"].as<int>();
+  if (options.count("maxdebt"))
+    max_debt = options["maxdebt"].as<int>();
+  if (options.count("daemon"))
+    run_as_daemon = true;
+  //  Backend Settings
+  if (options.count("backend") && options["backend"].as<std::string>() != BACKEND_TYPE)
+    backend_type = options["backend"].as<std::string>();
+  if (options.count("file"))
+    staticxml_path = options["file"].as<std::string>();
+  override_apidb_server_info("", &default_server_info, options);
+  oauth_server_info = override_apidb_server_info("oauth-", oauth_server_info, options);
+  update_server_info = override_apidb_server_info("update-", update_server_info, options);
+  if (options.count("disable-api-write"))
+    disable_api_write = true;
+  if (options.count("cachesize") && options["cachesize"].as<size_t>() != CHANGESET_CACHE_SIZE)
+    changeset_cache_size = options["cachesize"].as<size_t>();
+}
+
+apidb_server_info *Options::override_apidb_server_info(
+  const std::string &prefix,
+  apidb_server_info *info,
+  boost::program_options::variables_map &options) {
+  bool found = false;
+  apidb_server_info overrides;
+  if (options.count(prefix + "charset")) {
+    overrides.charset = options[prefix + "charset"].as<std::string>();
+    //  Because the 'charset' has a default value in 'options' it shouldn't
+    //  trigger an update of all default values overridden here
+    found = (overrides.charset != DEFAULT_CHARSET || info != &default_server_info);
+  }
+  if (options.count(prefix + "host")) {
+    overrides.host = options[prefix + "host"].as<std::string>();
+    found = true;
+  }
+  if (options.count(prefix + "dbport")) {
+    overrides.port = options[prefix + "dbport"].as<std::string>();
+    found = true;
+  }
+  if (options.count(prefix + "dbname")) {
+    overrides.dbname = options[prefix + "dbname"].as<std::string>();
+    found = true;
+  }
+  if (options.count(prefix + "username")) {
+    overrides.username = options[prefix + "username"].as<std::string>();
+    found = true;
+  }
+  if (options.count(prefix + "password")) {
+    overrides.password = options[prefix + "password"].as<std::string>();
+    found = true;
+  }
+  if (found) {
+    if (prefix.empty()) {
+      default_server_info = overrides;
+    } else if (&default_server_info == info) {
+      apidb_server_info *new_info = new apidb_server_info();
+      *new_info = overrides;
+      return new_info;
+    }
+    else {
+      *info = overrides;
+    }
+  }
+  return info;
+}
+
+std::string Options::get_configuration_options() const
+{
+  std::ostringstream ostr;
+  ostr << std::left << std::boolalpha;
+  std::_Setw w = std::setw(25);
+  /** General Settings */
+  ostr << "--- General Settings ---" << std::endl
```

Comment:
I left out this method as I currently don’t see a need for it. I think maybe there’s some easy way to let boost program options do something similar.